services/can_logger/can_logger.py

```diff
@@ -8,51 +7,124 @@
 import sys
 import collections
 import multiprocessing as mp
+import logging
 from datetime import datetime
 from psycopg2 import OperationalError
 from psycopg2.extras import execute_values
 
-
 def write_to_csv(wr_buff, can_bus):
-
     # Write to CSV from write buffer. First item is timestamp, second is
-    # can_id and third is can_data
-
-    with open(f'/data/log/{can_bus}.csv', mode='a') as logfd:
-
-        logcsv = csv.writer(logfd, delimiter=',', quotechar='"',
-                            quoting=csv.QUOTE_MINIMAL)
-
-        logcsv.writerows(wr_buff)
+    # can_bus, third is can_id, and last is can_data
+    # Generate timestamp to rotate the log
+    global log_path
+    #timestamp = rotate_log()
+    while(not log_path):
+        try:
+            with open(f'/data/log/can/{can_bus}.csv', mode='a') as logfd:
```

Comment:
This should either not be hard coded or a simple path. Maybe both?

default /data, overridden by `DATA_DIR=<path>` or `--data-dir=<path>` or something

The reason is the service should not know anything about the host system. It is the only thing in the container and so it "owns" the data folder. If the deployment wants files logged to the host filesystem, then they will map their path of choice to /data in the container or (more correctly due to user id / permissions issues with local FS mapping) just map /data to a volume.

Reply:
Indeed. This can be further included for a future NATS-based version.